modules/caddyhttp/app.go

```diff
@@ -343,6 +343,11 @@ func (app *App) Start() error {
 				// enable TLS if there is a policy and if this is not the HTTP port
 				useTLS := len(srv.TLSConnPolicies) > 0 && int(listenAddr.StartPort+portOffset) != app.httpPort()
 				if useTLS {
+					// create HTTP redirect wrapper, which detects if
+					// the request had HTTP bytes on the HTTPS port, and
+					// triggers a redirect if so.
+					ln = NewHttpRedirectListener(ln)
```

Comment:
This isn't used outside the package, let's unexport it.

And actually, it doesn't really need to be a function at all. Simply replace with:

```suggestion
					ln = &httpRedirectListener{Listener: ln}
```